File: src/main/java/facade/account/Account.java
package facade.account;

public abstract class Account {
    private long balance;

    protected Account(long balance) {
        this.balance = balance;
    }

    public abstract void deposit(long amount);

    public abstract void withdraw(long amount);

    public long getBalance() {
        return balance;
    }

    protected void addBalance(long amount) {
        balance += amount;
    }

    protected void minusBalance(long amount) {
        if (balance - amount < 0) {
            throw new IllegalArgumentException("잔고 부족");
        }
        balance -= amount;
    }
}

File: src/main/kotlin/facade/account/Account.kt
package facade.account

abstract class Account(private var balance: Long) {
    abstract fun deposit(amount: Long)
    abstract fun withdraw(amount: Long)

    fun getBalance(): Long {
        return balance
    }

    protected fun addBalance(amount: Long) {
        balance += amount
    }

    protected fun minusBalance(amount: Long) {
        if (balance - amount < 0) {
            throw IllegalArgumentException("잔고 부족")
        }

        balance -= amount
    }
}